architecture/core_concepts/haproxy-router.adoc

```diff
@@ -0,0 +1,76 @@
+[[architecture-core-concepts-haproxy-router]]
+= HAProxy Router Plug-in
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+
+A router is one way to
+xref:../../dev_guide/getting_traffic_into_cluster.adoc#getting-traffic-into-cluster[get
+traffic into the cluster]. The HAProxy Template Router plug-in is one of the
+available
+xref:../../architecture/core_concepts/routes.adoc#available-router-plug-ins[router
+plugins].
+
+[[haproxy-template-router]]
+== HAProxy Template Router
+
+The template router has two components:
+
+* a wrapper that watches endpoints and routes and causes a reload based on changes.
+* a controller that builds the HAProxy configuration file based on routes and endpoints.
+
+[NOTE]
+====
+The link:https://cbonte.github.io/haproxy-dconv/1.5/configuration.html[HAProxy router] uses version 1.5.18.
+====
+
+The controller and HAProxy are housed inside a pod, which is managed by a
+deployment configuration. The process of
+xref:../../install_config/router/index.adoc#[setting up the router] is automated
+by the `oadm router` command.
+
+The controller watches the routes and endpoints for changes, as well as
+HAProxy's health. When a change is detected, it builds a new haproxy-config file
+and restarts  HAProxy. The haproxy-config file is constructed based on the
+router's template file and  information from {product-title}. 
+
+The  HAProxy template file can be
+xref:../../install_config/router/customized_haproxy_router.adoc#[customized] as
+needed to support features that are not currently supported by {product-title}.
+The link:https://cbonte.github.io/haproxy-dconv/1.5/configuration.html#[HAProxy
+manual] describes all of the features supported by HAProxy.
+
+The following diagram illustrates how data flows from the master through the
+plug-in and finally into an HAProxy configuration:
+
+.HAProxy Router Data Flow
+image::router_model.png[HAProxy Router Data Flow]
+
+*Sticky Sessions*
+
+Implementing sticky sessions is up to the underlying router configuration. The
+default HAProxy template implements sticky sessions using the `balance source`
+directive, which balances based on the source IP. In addition, the template
+router plug-in provides the service name and namespace to the underlying
+implementation. This can be used for more advanced configuration, such as
+implementing stick-tables that synchronize between a set of peers.
+
+Specific configuration for this router implementation is stored in the
+*_/var/lib/haproxy/conf/haproxy-config.template_* file of the router container.
```

Comment:
@pecameron From our guidelines, filenames use underscores and asterisks:

https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference